storm-core/src/jvm/org/apache/storm/state/KeyValueState.java

```diff
@@ -45,4 +47,9 @@
      * @return the value or defaultValue if no mapping is found
      */
     V get(K key, V defaultValue);
+
+    /**
+     * @return all key value mappings as an unmodifiable map.
+     */
+    Map<K, V> getAll();
```

Comment:
>You mean an iterator over the original state? 

yes it can be an `Iterator<Map.Entry<K, V>>`. 

the issue with map is that you need to load the entire state into the map before you can return and it will not scale if the state is huge.
 
>iterator also has a remove method which shouldn't be a supported operation 

remove may be good to have, but to keep it simple can throw an UnsupportedOperationException